tests/foreman/api/test_ansible.py

```diff
@@ -123,3 +123,70 @@ def test_positive_ansible_job_on_host(target_sat, module_org, rhel_contenthost):
     )
     result = target_sat.api.JobInvocation(id=job['id']).read()
     assert result.succeeded == 1
+
+
+@pytest.mark.no_containers
+@pytest.mark.rhel_ver_match('9')
+def test_positive_ansible_job_on_multiple_host(
+    target_sat, module_org, rhel_contenthost, rhel7_contenthost, rhel8_contenthost
+):
+    """Test execution of Ansible job on multiple hosts simultaneously.
+
+    :id: 9369feef-466c-40d3-9d0d-65520d7f21ef
+
+    :customerscenario: true
+
+    :steps:
+        1. Register multiple content hosts with satellite
+        2. Import a role into satellite
+        3. Assign that role to all host
+        4. Trigger ansible job keeping all host in a single query
+        5. Check the passing and failing of individual hosts
+        6. Check if one of the job on a host is failed resulting into whole job is marked as failed.
+
+    :expectedresults:
+        1. One of the jobs failing on a single host must impact the overall result as failed.
+
+    :BZ: 2167396,2190464,2184117
+
+    :CaseAutomation: Automated
+    """
+    hosts = [rhel_contenthost, rhel8_contenthost, rhel7_contenthost]
+    SELECTED_ROLE = 'RedHatInsights.insights-client'
+    for _ in hosts:
+        _.install_katello_ca(target_sat)
+        _.register_contenthost(module_org.label, force=True)
+        assert _.subscribed
+        _.add_rex_key(satellite=target_sat)
+        id = target_sat.nailgun_smart_proxy.id
+        target_host = _.nailgun_host
+        target_sat.api.AnsibleRoles().sync(data={'proxy_id': id, 'role_names': [SELECTED_ROLE]})
+        target_sat.cli.Host.ansible_roles_assign(
+            {'id': target_host.id, 'ansible-roles': SELECTED_ROLE}
+        )
+        host_roles = target_host.list_ansible_roles()
```

Comment:
can we use API endpoints for assigning and listing ansible roles assigned?

Reply:
I believe for listing we are using the API endpoint only and for assigning I wanted to do it via the name of the role,
else I have to fetch the ansible_role_id explicitly again as previously we were doing it by hardcoding the id as 1, which was not working properly when the roles were re-assigned in automation.